imagebuildah/build.go

```diff
@@ -1047,7 +1066,12 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) error
 		if err := stageExecutor.Prepare(ctx, stage.Builder, stage.Node, ""); err != nil {
 			return err
 		}
-		defer stageExecutor.Delete()
+		// Always remove intermediate containers if either b.forceRmIntermediateCtrs is true
+		// or if building without layers.
+		// Will remove intermediate containers even if build fails.
+		if b.forceRmIntermediateCtrs || (!b.layers && !b.noCache) {
```

Comment:
I think b.layers and this flag should conflict, so this situation would not happen.
--layers=false --rm 
--layers=false --force-rm
Should return an error

Not sure about the b.noCache flag here?  Doesn't this mean the user chose to not use the cache during a rebuild?  Wouldn't he still want the layers removed?

Reply:
Right, so if we are not building with layers, this if condition shouldn't be true as the removal of the container will be handled by the defer. This is to ensure that if a build fails and --force-rm=false then the container where the failure occurred persists.

Reply:
Reworded the comment, hopefully it is clearer now.